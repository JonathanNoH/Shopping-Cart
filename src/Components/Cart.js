import React, { useState } from "react";

const Cart = ({cart, handleRemoveItem}) => {
  const [toggleCartDisplay, setToggleCartDisplay] = useState(false);
  const handleClick = () => {
    setToggleCartDisplay(prevState => !prevState);
  }
  
  let cartArray;
  cartArray = Object.entries(cart).map((item) => {
    return (
    <li key={item[0].toString()}>
      <div>{item[0]}</div>
      <div>{item[1]}</div>
      <button 
        onClick={() => {handleRemoveItem(item[0])}}
        aria-label={`Remove ${item[0]} from cart`}>Remove</button>
    </li>
    )
  })

  const cartTotal = Object.values(cart).reduce(
    (previous, current) => previous + current, 0
  );

  return (
    <div>
      <div>
        <button onClick={handleClick}>Cart</button>
        <span>{"("+ cartTotal +")"}</span>  
      </div>
      {toggleCartDisplay && 
      ((cartArray.length > 0) ? 
        cartArray : "Your cart is empty")
      }
    </div>
  )
}

export default Cart;