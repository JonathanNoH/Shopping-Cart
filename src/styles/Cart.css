.Cart {
  display: grid;
  grid-template-columns: max-content 1fr;
  gap: 1rem;
  padding: 1rem;
}

.showCart {
  grid-column: 1;
  grid-row: 1;
}

.cartItems {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(max-content, 300px));
  place-items: start;
}