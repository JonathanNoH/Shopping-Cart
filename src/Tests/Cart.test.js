import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";  // optional
import userEvent from "@testing-library/user-event";
import Cart from "../Components/Cart";

describe('Cart Component', () => {
  it("renders cart", () => {
    const { container } = render(<Cart cart={{}}/>);
    expect(container).toMatchSnapshot();
  })
});

describe('Accepts props', () => {
  it("displays empty cart message", () => {
    render(<Cart cart={{}}/>)

    //toggle to show cart
    userEvent.click(screen.getByRole('button', { name: 'Cart'}));

    expect(screen.getByText(/Your cart is empty/)).toBeInTheDocument();
  })
  it("displays items when cart is not empty", () => {
    render(<Cart cart={{Apple: 12, Orange: 6}}/>);

    //toggle to show cart
    userEvent.click(screen.getByRole('button', { name: 'Cart'}));

    expect(screen.getByText(/Apple/)).toBeInTheDocument();
    expect(screen.getByText(/12/)).toBeInTheDocument();
    expect(screen.getByText(/Orange/)).toBeInTheDocument();
    expect(screen.getByText(/6/)).toBeInTheDocument();
  })
})

it("button toggles viewing cart", () => {
  render(<Cart cart={{}}/>);
  const button = screen.getByRole('button', { name: 'Cart'});

  expect(screen.queryByText(/Your cart is empty/)).not.toBeInTheDocument();

  userEvent.click(button);
  expect(screen.getByText(/Your cart is empty/)).toBeInTheDocument();

  userEvent.click(button);
  expect(screen.queryByText(/Your cart is empty/)).not.toBeInTheDocument();
})

describe("cart total", () => {
  it("displays cart total for empty cart", () => {
    render(<Cart cart={{}} />)
    expect(screen.getByText(/0/)).toBeInTheDocument();
  })
  it("displays cart total for nonempty cart", () => {
    render(<Cart cart={{Apple: 12, Orange: 6}}/>);
    expect(screen.getByText(/18/)).toBeInTheDocument();
  })
})
