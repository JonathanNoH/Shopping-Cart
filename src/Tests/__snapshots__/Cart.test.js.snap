// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component renders cart 1`] = `
<div>
  <div>
    <div>
      <button>
        Cart
      </button>
      <span>
        (0)
      </span>
    </div>
  </div>
</div>
`;
